cfg/Instructions.h

```diff
@@ -276,14 +273,16 @@ INSN(TAbsurd) : public Instruction {
     std::string toString(const core::GlobalState &gs, const CFG &cfg) const;
     std::string showRaw(const core::GlobalState &gs, const CFG &cfg, int tabs = 0) const;
 };
-CheckSize(TAbsurd, 32, 8);
+CheckSize(TAbsurd, 24, 8);
 
 class InstructionPtr final {
     using tagged_storage = uint64_t;
 
-    static constexpr tagged_storage TAG_MASK = 0xffff;
-
-    static constexpr tagged_storage PTR_MASK = ~TAG_MASK;
+    static constexpr tagged_storage TAG_MASK = 0xff;
+    static constexpr tagged_storage FLAG_MASK = 0xff00;
+    static constexpr tagged_storage SYNTHETIC_FLAG = 0x0100;
+    static_assert((TAG_MASK & FLAG_MASK) == 0, "no bits should be shared between tags and flags");
+    static constexpr tagged_storage PTR_MASK = ~(FLAG_MASK | TAG_MASK);
```

Comment:
We do this for a couple of reasons:

1. A mask to compute the pointer would be a 48-bit (or so) mask, which is an inconvenient size for a constant on x86-64, requiring either an 11-byte (!) instruction to load the constant or a memory reference for the constant.  (A 48-bit all-ones mask would be OK on ARM64, since ARM64 has a reasonable suite of bit munging instructions).
2. Storing the tag in the low byte means that extracting the tag + actual pointer is two instructions and two registers:

```
movzx tagged_ptr_low_byte_register, tag
shr 16, tagged_ptr
# tagged_ptr now holds the actual pointer
```

(`movzx` can extract from the low 16 bits, too)

whereas putting the tag in the upper part would require more instructions:

```
mov tagged_ptr, temp
shr 48, temp # now holds the tag
# still need to mask tagged_ptr
```

so codesize is slightly smaller as a result.
3. Technically, pointers on x86-64  need to have their top 17 bits the same, and putting the pointer in the upper 48 bits enables us to adhere to that by using a signed right shift to extract the pointer.  Storing the pointer in the low 48 bits means we would need two shifts or gnarly conditional logic.  (I don't think we actually care about this in our tagged pointer implementation(s), but the constraint is there.)